guide/interactions/select-menus.md

```diff
@@ -0,0 +1,254 @@
+
+	message.awaitMessageComponentInteraction(filter, { time: 15000 })
+		.then(collected => console.log(`Collected ${collected.size} interactions`))
+		.catch(console.error);
+});
+```
+
+
+## Responding to select menus
+
+The `MessageComponentInteraction` class has similar methods as the `CommandInteraction` class, which we we'll be covering in the following section:
+
+::: tip
+The following methods behave exactly the same as on the `CommandInteraction` class:
+- `reply()`
+- `editReply()`
+- `defer()`
+- `fetchReply()`
+- `deleteReply()`
+- `followUp()`
+:::
+
+### Updating the select menu's message
+
+The `MessageComponentInteraction` class provides a method to update the message the button is attached to, by using `MessageComponentInteraction#update()`. We'll be passing an empty array as components, which will remove the menu after selecting an option:
+
+```js {1,3}
+client.on('interaction', async interaction => {
+	if (!interaction.isMessageComponent() && interaction.componentType !== 'SELECT_MENU') return;
+	if (interaction.customID === 'primary') await interaction.update('Something was selected', { components: [] });
+});
+```
+
+### Deferring and updating the select menu's message
+
+Additionally to deferring the response of the interaction, you can defer the button, which will trigger a loading state:
+
+```js {1,5-9}
+const wait = require('util').promisify(setTimeout);
+
+client.on('interaction', async interaction => {
+	if (!interaction.isMessageComponent() && interaction.componentType !== 'SELECT_MENU') return;
+	if (interaction.customID === 'primary') {
+		await interaction.deferUpdate();
+		await wait(4000);
+		await interaction.update('Something was selected!', { components: [] });
+	}
+});
+```
+
+
+## Multi-select menus
+
+A select menu is not bound to only one selection. You can specify a min and max amount of options that have to be selected. You can use `MessageSelectmMenu()#setMinValues()` and `MessageSelectMenu()#setMaxValues()` to select the minimumand maximum of required options to be selected by the user:
+
+```js {1,7-29,31}
+const { MessageActionRow, MessageSelectMenu } = require('discord.js');
+
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const row = new MessageActionRow()
+			.addComponents(new MessageSelectMenu()
+				.setCustomID('select')
+				.setPlaceholder('Nothing selected')
+				.setMinValues(2)
+				.setMaxValues(3)
+				.addOptions([
+					{
+						label: 'Select me',
+						description: 'This is a description',
+						value: 'first_selection',
+					},
+					{
+						label: 'You can select me too',
+						description: 'This is also a description',
+						value: 'second_selection',
+					},
+					{
+						label: 'I am also an option',
+						description: 'This is a description as well',
+						value: 'third_selection',
+					},
+				]));
+
+		await interaction.reply('Pong!', row);
```

Comment:
Should this be `{ components: [row] }` instead of `row`?

Reply:
Yes, my bad on that part, I had assumed the overload was valid like that, but it isn't